File: Python3/프로그래머스/1/178871. 달리기 경주/달리기 경주.py
def solution(players, callings):
    answer = {player: i for i, player in enumerate(players)}
    
    for i in callings:
        index = answer[i]
        answer[i] -= 1
        answer[players[index-1]] += 1
        players[index - 1], players[index] = players[index], players[index-1]
        
    return players

File: Swift/프로그래머스/1/178871. 달리기 경주/달리기 경주.swift
import Foundation

func solution(_ players:[String], _ callings:[String]) -> [String] {
     var players = players
    
    var answer: [String: Int] = [:]
    for (i, player) in players.enumerated() {
        answer[player] = i
    }
    
    for name in callings {
        if let index = answer[name], index > 0 {
            players.swapAt(index, index - 1)
            
            answer[name] = index - 1
            
            let frontPlayer = players[index] 
            answer[frontPlayer] = index
        }
    }
    
    return players
}